File: app/src/main/java/ru/mrlargha/stemobile/ui/substitutionadd/SubstitutionAddFormStateBuilder.java
package ru.mrlargha.stemobile.ui.substitutionadd;

public class SubstitutionAddFormStateBuilder {
    private String mDateError = null;
    private String mCabinetError = null;
    private String mGroupError = null;
    private String mPairError = null;
    private String mTeacherError = null;
    private String mSubjectError = null;
    private boolean mHasErrors = false;
    private String mCustomError = null;

    public String getDateError() {
        return mDateError;
    }

    public SubstitutionAddFormStateBuilder setDateError(String dateError) {
        mDateError = dateError;
        mHasErrors = true;
        return this;
    }

    public String getCabinetError() {
        return mCabinetError;
    }

    public SubstitutionAddFormStateBuilder setCabinetError(String cabinetError) {
        mCabinetError = cabinetError;
        mHasErrors = true;
        return this;
    }

    public String getGroupError() {
        return mGroupError;
    }

    public SubstitutionAddFormStateBuilder setGroupError(String groupError) {
        mGroupError = groupError;
        mHasErrors = true;
        return this;
    }

    public String getPairError() {
        return mPairError;
    }

    public String getTeacherError() {
        return mTeacherError;
    }

    public SubstitutionAddFormStateBuilder setTeacherError(String teacherError) {
        mTeacherError = teacherError;
        mHasErrors = true;
        return this;
    }

    public String getSubjectError() {
        return mSubjectError;
    }

    public SubstitutionAddFormStateBuilder setSubjectError(String subjectError) {
        mSubjectError = subjectError;
        mHasErrors = true;
        return this;
    }

    public boolean hasErrors() {
        return mHasErrors;
    }

    public SubstitutionAddFormStateBuilder setHasErrors(boolean hasErrors) {
        mHasErrors = hasErrors;
        return this;
    }

    public SubstitutionAddFormState createSubstitutionAddFormState() {
        return new SubstitutionAddFormState(mDateError, mCabinetError, mGroupError, mTeacherError,
                mSubjectError, mCustomError, mHasErrors);
    }

    public String getCustomError() {
        return mCustomError;
    }

    public SubstitutionAddFormStateBuilder setCustomError(String customError) {
        mCustomError = customError;
        mHasErrors = true;
        return this;
    }
}

File: app/src/main/java/ru/mrlargha/stemobile/ui/substitutionadd/SubstitutionAddFormStateBuilder.kt
package ru.mrlargha.stemobile.ui.substitutionadd

class SubstitutionAddFormStateBuilder {
    var dateError: String? = null
        private set
    var cabinetError: String? = null
        private set
    var groupError: String? = null
        private set
    val pairError: String? = null
    var teacherError: String? = null
        private set
    var subjectError: String? = null
        private set
    private var mHasErrors = false
    var customError: String? = null
        private set

    fun setDateError(dateError: String?): SubstitutionAddFormStateBuilder {
        this.dateError = dateError
        mHasErrors = true
        return this
    }

    fun setCabinetError(cabinetError: String?): SubstitutionAddFormStateBuilder {
        this.cabinetError = cabinetError
        mHasErrors = true
        return this
    }

    fun setGroupError(groupError: String?): SubstitutionAddFormStateBuilder {
        this.groupError = groupError
        mHasErrors = true
        return this
    }

    fun setTeacherError(teacherError: String?): SubstitutionAddFormStateBuilder {
        this.teacherError = teacherError
        mHasErrors = true
        return this
    }

    fun setSubjectError(subjectError: String?): SubstitutionAddFormStateBuilder {
        this.subjectError = subjectError
        mHasErrors = true
        return this
    }

    fun hasErrors(): Boolean {
        return mHasErrors
    }

    fun setHasErrors(hasErrors: Boolean): SubstitutionAddFormStateBuilder {
        mHasErrors = hasErrors
        return this
    }

    fun createSubstitutionAddFormState(): SubstitutionAddFormState {
        return SubstitutionAddFormState(dateError, cabinetError, groupError, teacherError,
                subjectError, customError, mHasErrors)
    }

    fun setCustomError(customError: String?): SubstitutionAddFormStateBuilder {
        this.customError = customError
        mHasErrors = true
        return this
    }
}